src/xcode/ENA/ENA/Source/Scenes/ContactDiary/StoreV2/ContactDiaryStoreV2.swift

```diff
@@ -0,0 +1,977 @@
+	}
+
+	@discardableResult
+	func reset() -> DiaryStoringVoidResult {
+		let dropTablesResult = dropTables()
+		if case let .failure(error) = dropTablesResult {
+			return .failure(error)
+		}
+
+		let openAndSetupResult = openAndSetup()
+		if case .failure = openAndSetupResult {
+			return openAndSetupResult
+		}
+
+		var updateDiaryDaysResult: DiaryStoringVoidResult?
+		databaseQueue.inDatabase { database in
+			updateDiaryDaysResult = updateDiaryDays(with: database)
+		}
+		if case .failure(let error) = updateDiaryDaysResult {
+			return .failure(error)
+		}
+
+		return .success(())
+	}
+
+	func close() {
+		databaseQueue.close()
+	}
+
+	// MARK: - Private
+
+	private let dataRetentionPeriodInDays = 16 // Including today.
+	private let userVisiblePeriodInDays = 14 // Including today.
+	private let key: String
+	private let dateProvider: DateProviding
+	private let schema: ContactDiaryStoreSchemaV2
+
+	private var todayDateString: String {
+		dateFormatter.string(from: dateProvider.today)
+	}
+
+	private var dateFormatter: ISO8601DateFormatter = {
+		let dateFormatter = ISO8601DateFormatter.contactDiaryFormatter
+		return dateFormatter
+	}()
+
+	private var germanDateFormatter: DateFormatter = {
+		let dateFormatter = DateFormatter()
+		dateFormatter.dateStyle = .short
+		dateFormatter.timeStyle = .none
+		dateFormatter.locale = Locale(identifier: "de_DE")
+		return dateFormatter
+	}()
+
+	private let databaseQueue: FMDatabaseQueue
+	private let latestVersion: Int
+
+	private func openAndSetup() -> DiaryStoringVoidResult {
+		var errorResult: DiaryStoringVoidResult?
+		
+		databaseQueue.inDatabase { database in
+			Log.info("[ContactDiaryStore] Open and setup database.", log: .localData)
+			
+			let dbHandle = OpaquePointer(database.sqliteHandle)
+			guard CWASQLite.sqlite3_key(dbHandle, key, Int32(key.count)) == SQLITE_OK else {
+				Log.error("[ContactDiaryStore] Unable to set Key for encryption.", log: .localData)
+				errorResult = .failure(dbError(from: database))
+				return
+			}
+			
+			guard database.open() else {
+				Log.error("[ContactDiaryStore] Database could not be opened", log: .localData)
+				errorResult = .failure(dbError(from: database))
+				return
+			}
+			// We need to check for the following:
+			// (userVersion = 0) means: no database was created and stored previously by the user --> this is the first database --> create new updated schema
+			// if a database was stored but not the latest version --> migrate the old schema
+			if database.userVersion != 0 && database.userVersion < latestVersion {
+				// we create the migration here and not in the init() because the database need to be open and decrypted before migration or it will fail
+				// we added the migration twice because there is a bug here; initial database version should be 0 not 1, this is a temp fix
+				let migrations: [Migration] = [ContactDiaryMigration1To2(database: database), ContactDiaryMigration1To2(database: database)]
```

Comment:
Adding the migration twice doesn't fix the root problem.
Better fix the root problem by decouple access to the migration list from the version inside of the migrator implementation.